astropy/table/table.py

```diff
@@ -2717,7 +2717,8 @@ def to_pandas(self):
             raise ValueError("Cannot convert a table with mixin columns to a pandas DataFrame")
 
         if any(getattr(col, 'ndim', 1) > 1 for col in self.columns.values()):
-            raise ValueError("Cannot convert a table with multi-dimensional columns to a pandas DataFrame")
+            badcols = [name for name,col in self.columns.items() if (getattr(col, 'ndim', 1) > 1)]
```

Comment:
Move this out in front of the original `if` block e.g.:
```
badcols = [name for name,col in self.columns.items() if (getattr(col, 'ndim', 1) > 1)]
if badcols:
    ...
```
Also, please keep the line width < 90 characters and adhere to PEP8 (in particular space after `,`).